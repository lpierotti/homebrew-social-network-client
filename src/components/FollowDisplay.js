import React from 'react'
import { Link } from 'react-router-dom'
import { Card, Image } from 'semantic-ui-react'


const FollowDisplay = (props) => {

	return (
			<Card raised={true}>
				<div style={{margin: 'auto'}}>
					{props.data.image ? <Link to={`/user/${props.data.id}/profile`}><Image className='smallDisplay' src={props.data.image} alt='' centered={true}/></Link> : <Link to={`/user/${props.data.id}/profile`}><Image className='smallDisplay' src='/default-profile.png' alt='' centered={true}/></Link>}
					<Link to={`/user/${props.data.id}/profile`}>{props.data.username}</Link>
				</div>
			</Card>
	)
}

export default FollowDisplay