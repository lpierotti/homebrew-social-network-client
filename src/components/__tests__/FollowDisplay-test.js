import React from 'react';
import { shallow, mount, render } from 'enzyme';
import FollowDisplay from '../FollowDisplay';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom'

describe('FollowDisplay component', () => {

	it('renders correctly', () => {
		const tree = renderer.create(
			<StaticRouter context={{}}><FollowDisplay data={{}}/></StaticRouter>
		).toJSON();
		expect(tree).toMatchSnapshot();
	})
})