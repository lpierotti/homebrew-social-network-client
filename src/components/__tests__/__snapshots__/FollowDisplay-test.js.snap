// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowDisplay component renders correctly 1`] = `
<div
  className="ui raised card"
  href={undefined}
  onClick={[Function]}
>
  <div
    style={
      Object {
        "margin": "auto",
      }
    }
  >
    <a
      href="/user/undefined/profile"
      onClick={[Function]}
    >
      <img
        alt=""
        className="ui centered image smallDisplay"
        height={undefined}
        src="/default-profile.png"
        width={undefined}
      />
    </a>
    <a
      href="/user/undefined/profile"
      onClick={[Function]}
    />
  </div>
</div>
`;
